### DISCUSSION TITLE
Opti stack in MPC trajectory tracking

### DISCUSSION THREAD

**Discussion by snDevX** [casadi/casadi#3770] [Q&A] [answered]
Hello,

I am using opti stack to solve a trajectory tracking problem.
How can I use the optimal input sequence generated at one iteration to update my reference input (u_ref) in my cost, for the next iteration?

**Comment by sandeep026** [comment, marked as answer]
https://groups.google.com/g/casadi-users/c/MMK9iR84mpo/m/ViH5CYmpEQAJ has a sample code to implement MPC using optistack for reference tracking

**Comment by snDevX** [comment]
that helped, thanks
